### ISSUE TITLE
Drag Drop Chapter in Outline

### ISSUE THREAD

**Issue by gruenwaldt** [typora/typora-issues#916]
It would be nice to make it possible to reorder chapter by dragging and dropping in the outline window.

**Comment by shoesforindustry** [comment]
What a great idea +1

**Comment by hodanli** [comment]
this would be great. makes the typora more complete text editor.

**Comment by dadoprom** [comment]
so when will this feature be implemented?

**Comment by aourednik** [comment]
Yes, this would make a __huge__ difference, indeed. A very strange thing is that the only apps that support reordering chapters in any type of document via outline are MS Word for Windows (the mac version does not) or LibreOffice (now broken on Mac for some High Sierra issue). Scrivener can do it but is overkill. No other app proposes the feature. So yes, this would make Typora really outstanding

**Comment by vassudanagunta** [comment]
#2376 is a related request to support structure manipulation (headings/chapters, table rows, list items, regular blocks).

**Comment by jediwarrior77** [comment]
This feature is a must for large documents, we need to be able to re-arrange outline chapters. I dont know how difficult it is to implement but I'm amazed by the lack of this no brainer feature

**Comment by jediwarrior77** [comment]
Guys Zettlr markup editor CAN do it!!! its also a visual editor, very similar to Typora.

**Comment by vassudanagunta** [comment]
> I dont know how difficult it is to implement but I'm amazed by the lack of this no brainer feature

Wow, that was extremely rude, entitled, and arrogant.

**Comment by jediwarrior77** [comment]
@vassudanagunta Another reptilian boy. I am not impressed by your lack of common sense and empty bucket snake-like criticism. Go fetch some water.

**Comment by jediwarrior77** [comment]
> > I dont know how difficult it is to implement but I'm amazed by the lack of this no brainer feature
> 
> Wow, @jediwarrior77, that was extremely rude, entitled, and arrogant.
@vassudanagunta Another reptilian boy. I am not impressed by your lack of common sense and empty bucket snake-like criticism. Go fetch some water.

**Comment by SaschaFast** [comment]
+1

One of the selling points of typora is the very awesome outline view. Outlining is in practice coupled with re-arranging the different parts of the text. So, improving the outline functionality by adding this aspect would be GREAT. :)

**Comment by FelixOlschewski** [comment]
+1

Typora is already pretty much the best of its class. Writing is about lining up your thoughts and getting them in the right order. It is only natural to move words in a sentence and rearrange sentences for maximum impact. Re-ordering sections/chapters is unavoidable when writing a book. The ability to do this via drag & drop would be a killer feature to have in Typora.

**Comment by wideawakening** [comment]
+1 IMHO; killer feature

**Comment by johnbede2000** [comment]
+1
Absolutely agree, this is a feature I want for organising notes and therefore organising & refining my thoughts.

At the moment I'm using the web app Workflowy.com to do this, by dragging around hierarchical bullet points.
But I'd love to do it inside Typora - I've downloaded the iA Writer Duo font and use that in Typora :)

**Comment by aourednik** [comment]
@paolopan83 could you say which ones? I have been looking for a Markdown editor capable to do this for years

**Comment by paolopan83** [comment]
Obisidian does it. But even Obsidian lacks the possbility to change heading level by drang and drop.

**Comment by K9PAL** [comment]
I would like to put in another plea for this feature. It's a mainstream use case to move outline items to new positions and I was actually expecting a feature to allow that, though the more I think about it, it could be harder than it looks.

May I suggest looking at some outlining products from the 90s, like Ecco and More, for ideas?

**Comment by SHANG-PENG** [comment]
+1
After dragging and reordering the titles, the corresponding contents will also be updated simultaneously. If so, it would be extremely great.
